File: src/test/scala/dregex/CanonicalEqTest.scala
package dregex

import java.util.regex.Pattern
import dregex.TestUtil.using
import org.junit.jupiter.api.Assertions.{assertFalse, assertTrue}
import org.junit.jupiter.api.Test

class CanonicalEqTest {

  @Test
  def testCanonicalEquivalence() = {

    using(Regex.compile("\u00F6")) { r =>
      assertTrue(r.matches("\u00F6"))
      assertFalse(r.matches("\u006F\u0308"))
    }

    using(Regex.compile("\u00F6", Pattern.CANON_EQ)) { r =>
      assertTrue(r.matches("\u00F6"))
      assertTrue(r.matches("\u006F\u0308"))
    }

  }

}

File: src/test/java/dregex/CanonicalEqTest.java
package dregex;

import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.regex.Pattern;
import org.junit.jupiter.api.Test;

class CanonicalEqTest {

    @Test
    void testCanonicalEquivalence() {
        assertTrue(Regex.compile("a\u030A", Pattern.CANON_EQ).matches("\u00E5"));
        assertTrue(Regex.compile("\u00E5", Pattern.CANON_EQ).matches("a\u030A"));
        assertTrue(Regex.compile("[\u00E5-\u00F0]", Pattern.CANON_EQ).matches("a\u030A"));
    }
}
